devices/flash-flexspi/fspi/fspi.c

```diff
@@ -0,0 +1,476 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * FlexSPI Controller driver
+ *
+ * Copyright 2021-2022 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include "../fspi.h"
+#include "../lut.h"
+
+
+/* Include platform dependent pin config */
+#if defined(IMXRT117X)
+#include "fspi_rt117x.h"
+#elif defined(IMXRT106X)
+#include "fspi_rt106x.h"
+#elif defined(IMXRT105X)
+#include "fspi_rt105x.h"
+#else
+#error "FlexSPI is not supported on this target"
+#endif
```

Comment:
Perhaps should be moved to board_config.h - later

Reply:
I agree.